### DISCUSSION TITLE
Pdfplumber Failing on a particular PDF.

### DISCUSSION THREAD

**Discussion by adicognext** [jsvine/pdfplumber#565] [Ask for help with specific PDFs]
## The bug

*Hey, I wanted to extract tables from a pdf using pdfplumber, I tried the default setting and with multiple join tolerances but I got empty tables only If anyone could suggest some settings It would be very helpful.*

## Code to reproduce the problem

*Paste it here, or attach a Python file.*
![Screenshot from 2021-12-14 16-18-27](https://user-images.githubusercontent.com/90697470/145984277-a4e5dbd7-9e56-4eac-8dae-997a3aefe088.png)

## PDF file

*Concerned PDF* 
[Navi 2019-20.pdf](https://github.com/jsvine/pdfplumber/files/7710970/Navi.2019-20.pdf)

## Actual behavior

*No tables were detected*

## Screenshots

*Concerned tables*
![page4](https://user-images.githubusercontent.com/90697470/145985026-4c694d52-5ed6-4f4c-9232-4ee2bb8d5418.jpg)
![page3](https://user-images.githubusercontent.com/90697470/145985032-672b8c7f-73ba-4f6d-8144-17bd3cc4fbd0.jpg)
![page2](https://user-images.githubusercontent.com/90697470/145985036-3999a2ed-a4a9-4ce8-9b18-1e47ce67d3a4.jpg)

## Environment

- pdfplumber version: 0.5.28
- Python version: 3.8.5
- OS: Linux

**Comment by jsvine** [comment]
Hi @adicognext, and thanks for your interest in this library. To investigate why a table might not be found, you can always use `page.to_image().debug_tablefinder(...)`. For instance, with your settings:

<img width="631" alt="Screen Shot 2021-12-14 at 9 51 57 AM" src="https://user-images.githubusercontent.com/534702/146021609-03d5fdf7-9e46-4c4e-a34a-e09b34db63d8.png">

Inspecting that, you can see that there are no vertical lines appearing and that, perhaps, you meant this instead, using `text` for the *vertical* setting and the `lines` for *horizontal* setting, rather than vice versa:

<img width="626" alt="Screen Shot 2021-12-14 at 9 54 27 AM" src="https://user-images.githubusercontent.com/534702/146022072-a9813cec-e094-4105-9a31-57f15ec4c29f.png">

That *does* find a table, but chops it up a bit finely, because of how many different alignments the text has, even within what presumably you'd call a single column. Instead, we can try to use the gaps between the dotted lines that seem to separate each column:

<img width="55" alt="Screen Shot 2021-12-14 at 9 55 24 AM" src="https://user-images.githubusercontent.com/534702/146022271-6c7e519b-fd91-4a23-94b9-3f7c8e64db0f.png">

Because there's some nuance to how those lines are specified in the PDF, it took a little bit of trial and error, but this seems to get fairly close to what I imagine you want:

```python
thin_lines = [ line for line in page.lines
    if line["linewidth"] == 0.5 and line["top"] < 300 ]


# This merges overlapping line segments, due to a
# quirk in the PDF's design.
thin_lines_joined = pdfplumber.table.join_edge_group(
    thin_lines,
    "h",
    tolerance=0,
)

im.reset().debug_tablefinder({
    "vertical_strategy": "explicit",
    "explicit_vertical_lines": (
        # Using both the left- and right-hand edges of each line
        [ x["x0"] for x in thin_lines_joined ] +
        [ x["x1"] for x in thin_lines_joined ]
    ),
    "horizontal_strategy": "lines",
    "join_tolerance": 50,
    "snap_tolerance": 5,
})
```

![image](https://user-images.githubusercontent.com/534702/146024077-f5451c21-58b4-43bd-96e7-f536b4ece711.png)

If you want to get rid of the errant cells at the top and bottom, you could use `cropped = page.crop(...)` first.

**Reply by adicognext** [reply]
Yes, my bad. Thank you so much for your reply It helped a lot.
